mllib/src/main/scala/org/apache/spark/ml/tree/impl/GradientBoostedTrees.scala

```diff
@@ -258,11 +258,13 @@ private[spark] object GradientBoostedTrees extends Logging {
     val baseLearnerWeights = new Array[Double](numIterations)
     val loss = boostingStrategy.loss
     val learningRate = boostingStrategy.learningRate
-    // Prepare strategy for individual trees, which use regression with variance impurity.
+    // Prepare strategy for individual trees, which all use regression.
+    // TODO(vlad17): Changing the strategy here is confusing (especially using regression for
```

Comment:
Anywhere you need to leave a TODO, it'd be good to create a corresponding JIRA to fix it, where that JIRA is linked as blocked by this current JIRA.  Here, you could just use ```TODO(SPARK-16728)```